products/memgraph/v0.51.0/reference_guide/query-modules.md

```diff
@@ -196,6 +200,123 @@ As previously mentioned, no exceptions should leave your module. If you are
 writing the module in a language which throws them, you probably want
 exception handlers in `mgp_init_module` and `mgp_shutdown_module` as well.
 
+#### Python API
+
+Query modules can be implemented using the Python API provided by Memgraph. If
+you wish to write your own query modules using Python API, you need to have
+Python version `3.5.0` and above.
+
+Let's take a look at the `py_example.py` file.
+
+```python
+import mgp
+```
+
+On the first line, we import the `mgp` module, which contains definitions of the
+public Python API provided by Memgraph. In essence, this is a wrapper around the
+C API described in the previous section. This file (`mgp.py`) can be found in
+the Memgraph installation directory, under `include/memgraph`. On the standard
+Debian installation, this will be under `/usr/include/memgraph`.
+
+Next we have a `procedure` function. This function will serve as the callback
+for our `py_example.procedure` invocation through openCypher.
+
+```python
+@mgp.read_proc
+def procedure(context: mgp.ProcCtx,
+              required_arg: mgp.Nullable[mgp.Any],
+              optional_arg: mgp.Nullable[mgp.Any] = None
+              ) -> mgp.Record(args=list,
+                              vertex_count=int,
+                              avg_degree=mgp.Number,
+                              props=mgp.Nullable[mgp.Map]):
+
+  ...
+```
+
+This procedure needs to be a callable which optionally takes `ProcCtx` as the
+first argument. Other arguments will be bound to values passed in the
+cypher query. The full signature of this procedure needs to be annotated
+with types. The return type must be `Record(field_name=type, ...)` and the
+procedure must produce either a complete `Record` or `None`. As you can see,
+the procedure is passed to a `read_proc` decorator which handles read-only
+procedures. You can also inspect the definition of said decorator in
+`mgp.py` file.
+
+In our case, the example procedure returns 4 fields:
+
+- `args`, a copy of arguments passed to the procedure.
+- `vertex_count`, number of vertices in the database.
+- `avg_degree`, average degree of vertices.
+- `props`, properties map of the Vertex or Edge object passed in `required_arg`.
+   In case a Path instance is passed, procedure returns the properties map of
+   the starting vertex.
+
+This procedure can be invoked in openCypher as follows:
+
+```opencypher
+MATCH (n) CALL py_example.procedure(n, 1) YIELD * RETURN *;
```

Comment:
I would change this to `MATCH (n) WITH n LIMIT 1 CALL py_example.procedure(n, 1) YIELD * RETURN *;`